src/app/left-menu/editor/editor.component.ts

```diff
@@ -11,10 +12,20 @@ import { MetaItem } from '../../types/meta-item';
 export class EditorComponent {
 
     @Input('item') item: MetaItem;
+    @Input() searchText = '';
 
-    constructor(private names: NamesService, private metaRepository: MetadataRepositoryService) { }
+    constructor(private names: NamesService,
+                private metaRepository: MetadataRepositoryService,
+                private sanitizer: DomSanitizer) { }
 
-    getRealName = (name) => this.names.getRealName(name);
+    highlight(text: string) {
+        text = this.names.getRealName(text);
+        if(this.searchText.length >= 3) {
+            text = '<span style="color:#f05b41">' + text.slice(0, this.searchText.length) + '</span>' + text.slice(this.searchText.length);
```

Comment:
Highlighting algorithm is completely wrong. Try to search any word not from the start of the string, 'color' for example.